spec/scenarios/dunning/campaign_v1_spec.rb

```diff
@@ -0,0 +1,232 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe "Dunning Campaign v1", :scenarios, type: :request do
+  let(:webhook_url) { "https://test.co/lago" }
+  let(:organization) do
+    create(:organization,
+      name: "JC AI",
+      premium_integrations: %w[auto_dunning],
+      email_settings: [],
+      webhook_url:)
+  end
+
+  let(:dunning_campaign) do
+    create(:dunning_campaign, organization:,
+      applied_to_organization: true, max_attempts: 2, days_between_attempts: 2)
+  end
+  let(:dunning_campaign_threshold) do
+    create(:dunning_campaign_threshold, dunning_campaign:, amount_cents: 150_00, currency: "EUR")
+  end
+  let(:stripe_cus_id) { "cus_123456789" }
+  let(:stripe_pm_id) { "pm_123456" }
+
+  let(:stripe_provider) { create(:stripe_provider, organization:) }
+  let(:customer) { create(:customer, organization:, payment_provider: :stripe, payment_provider_code: stripe_provider.code, net_payment_term: 2) }
+  let(:stripe_customer) { create(:stripe_customer, customer:, payment_provider: stripe_provider, provider_customer_id: stripe_cus_id) }
+
+  let(:external_subscription_id) { "sub_overdue-dunning-campaign-v1" }
+  let(:plan) { create(:plan, organization:, pay_in_advance: true, amount_cents: 149_00) }
+  let!(:addon) { create(:add_on, organization:) }
```

Comment:
do we need to create the add-on with the bang (!)... or wait until it is referenced... anyway, for `dunning_campaign_threshold` it is created in the before block... I would prefer to use one approach or the other but mixing both could be confusing overtime.

Reply:
Indeed! I moved it to the generation of the one_off invoice because it's not an important piece we reuse 👍